osu.Game/Screens/SelectV2/BeatmapCarousel.cs

```diff
@@ -113,7 +113,9 @@ private void beatmapSetsChanged(object? beatmaps, NotifyCollectionChangedEventAr
                         int previousIndex = Items.IndexOf(beatmap);
                         Debug.Assert(previousIndex >= 0);
 
-                        BeatmapInfo? matchingNewBeatmap = newSetBeatmaps.SingleOrDefault(b => b.DifficultyName == beatmap.DifficultyName && b.Ruleset.Equals(beatmap.Ruleset));
+                        BeatmapInfo? matchingNewBeatmap =
+                            newSetBeatmaps.SingleOrDefault(b => b.OnlineID == beatmap.OnlineID) ??
```

Comment:
Probably should ensure that both of these are actually `> 0` as well to avoid matching some complete garbage offline beatmap.